### PR TITLE
Component Status Board + Foundation for API-specific Tasks 

### PR DESCRIPTION
**Pull request by sghoweri**
image

## Jira
http://vjira2:8080/browse/BDS-1046 and http://vjira2:8080/browse/BDS-744

## Summary
Adds an automatically generated (always up to date) component status board to Pattern Lab's main overview page.

### Pattern Lab Updates
- Patches Pattern Lab PHP logic to add a brand new `--dataonly` CLI config option to export the global PL data available (ex. nav URLs) without having to do a full Pattern Lab build

### Build-related Updates
- Moves the existing `bolt-versions` build task to new api-tasks folder in `@bolt/build-tools`
- Adds a new api-specific section to the build tasks as a new home for handling things such as:
    - Generating the data / component checks for the component status board + compiling the Twig via `@bolt/twig-renderer`
    - Generating visual regression testing URLs
    - Programmatically being able to find all available versions of the Bolt packages published to NPM 
- Updates the existing pattern lab build task to conditionally allow generating just the PL data (quick compile)
- Consolidates + improve coordination of the build and watch tasks for the new API-specific work vs the build and watch tasks for Pattern Lab (via events that get emitted) 

### Misc
- Updates the `iframe-resizer` library import path (used in a few places) to prevent build errors (build broken otherwise)

## How to test
- Review changes to make sure we're all on the same page
- Verify Travis builds and deploys as expected and the main PL overview page correctly displays the new component explorer
- Confirm Pattern Lab's watch task will rebuild and compile as expected
- Confirm changes to the status board's API-specific task in the build tools will regenerate when you change the template / data + subsequently recompile Pattern Lab after completed

> NOTE: a handful of components displayed in the component status table (ex. UL, OL and Button Group) are going to be slightly mismatched with the actual url in Pattern Lab due to some naming differences + bug in Pattern Lab PHP's logic with folder names.

I'm getting a separate PR going that'll work around that by renaming the problematic PL folders to match up with the component name / name we publish to NPM with.

### CODE REVIEW

**Comment by sghoweri** [comment]
> Everything looks great. And i approve this PR. I notice one thing that on the first compile we have two `Recompiling Pattern Lab...` task triggered one after another. Maybe it should be that way but i want to notice that.
> image
> And then two `Pattern Lab recompiled in xxxms`. Recompiling times are not always the same.

Thanks @adamszalapski! I think I might just need to update the logic for this to make sure that compiling just the data for Pattern Lab doesn't count as "being the 1st time being compiled". Should be a super quick edit to make 👍

File: packages/build-tools/tasks/api-tasks/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+API tasks are a new (potentially emerging) pattern to help with gathering specific pieces of data that get collected and added to the Bolt Global Data store.
+
+### Examples of this include:
+- Determining what versions of Bolt are available (previously located in `@bolt/build-tools/utils)
```
Comment (danielamorse) [inline packages/build-tools/tasks/api-tasks/README.md]: Missing closing tick

File: docs-site/.patches/add-data-only-command--builder.patch [hunk 2]
```diff
@@ -0,0 +1,197 @@
+diff --git a/src/PatternLab/Builder.php b/src/PatternLab/Builder.php
```
Comment (danielamorse) [inline docs-site/.patches/add-data-only-command--builder.patch]: @sghoweri why all the `.patch` files?

File: packages/build-tools/tasks/api-tasks/README.md (hunk 1 above)
Comment (sghoweri) [reply packages/build-tools/tasks/api-tasks/README.md]: 👍 all set

File: docs-site/.patches/add-data-only-command--builder.patch (hunk 2 above)
Comment (sghoweri) [reply docs-site/.patches/add-data-only-command--builder.patch]: ^ These allow us to update Pattern Lab PHP Core with new features & functionality while still (mostly) usibg off the off the shelf code from Packagist. As an added bonus, this also makes it easier for us to package up and PR any updates that make sense and would benefit the community.

We use https://github.com/cweagans/composer-patches under the hood to help auto-patch what gets installed and attempt to apply our patches if everything works as expected

**Comment by sghoweri** [comment]
@danielamorse all set with those two things you pointed out. Anything else you wanted to call out?

**Pull request merged**